src/modules/did/did.processor.ts

```diff
@@ -1,92 +1,78 @@
 import { DIDService } from './did.service';
-import { ADD_DID_DOC_QUEUE_NAME, UPDATE_DID_DOC_QUEUE_NAME } from './did.types';
+import {
+  ADD_DID_DOC_JOB_NAME,
+  PIN_CLAIM_JOB_NAME,
+  PIN_CLAIM_QUEUE_NAME,
+  UPDATE_DID_DOC_JOB_NAME,
+  UPDATE_DOCUMENT_QUEUE_NAME,
+} from './did.types';
 
-@Processor('dids')
+@Processor(UPDATE_DOCUMENT_QUEUE_NAME)
 export class DIDProcessor {
-  private didCluster: DidStoreCluster;
-
   constructor(
     private readonly didService: DIDService,
     private readonly logger: Logger,
     private readonly configService: ConfigService,
-    private didInfura: DidStoreInfura
+    @InjectQueue(UPDATE_DOCUMENT_QUEUE_NAME)
+    private queue: Queue,
+    @InjectQueue(PIN_CLAIM_QUEUE_NAME)
+    private pinQueue: Queue
   ) {
     this.logger.setContext(DIDProcessor.name);
-
-    const IPFS_CLUSTER_ROOT = this.configService.get('IPFS_CLUSTER_ROOT');
-    const IPFS_CLUSTER_USER = this.configService.get('IPFS_CLUSTER_USER');
-    const IPFS_CLUSTER_PASSWORD = this.configService.get(
-      'IPFS_CLUSTER_PASSWORD'
-    );
-    const Authorization = `Basic ${Buffer.from(
-      `${IPFS_CLUSTER_USER}:${IPFS_CLUSTER_PASSWORD}`
-    ).toString('base64')}`;
-    this.didCluster = new DidStoreCluster(IPFS_CLUSTER_ROOT, {
-      Authorization,
-    });
   }
 
-  @OnQueueError()
+  @OnGlobalQueueError()
   onError(error: Error) {
     this.logger.error(error);
   }
 
-  @Process(ADD_DID_DOC_QUEUE_NAME)
-  public async processDIDDocumentAddition(job: Job<string>) {
-    try {
-      this.logger.debug(`processing cache add for ${job.data}`);
-      const doc = await this.didService.addCachedDocument(job.data);
+  @OnGlobalQueueActive()
+  async onActive(jobId: number) {
+    const job = await this.queue.getJob(jobId);
+    this.logger.debug(`Starting ${job.name} document ${job.data}`);
+  }
 
-      await this.pinClaims(doc.service.map((s) => s.serviceEndpoint)).catch(
-        (err) => {
-          this.logger.error(`error pinning the claim: ${err}`);
-        }
-      );
-    } catch (err) {
-      this.logger.error(`error adding ${job.data}: ${err}`);
-      throw err;
-    }
+  @OnGlobalQueueStalled()
+  async onStalled(jobId: number) {
+    const job = await this.queue.getJob(jobId);
+    this.logger.debug(`Stalled ${job.name} document ${job.data}`);
   }
 
-  @Process(UPDATE_DID_DOC_QUEUE_NAME)
-  public async processDIDDocumentRefresh(job: Job<string>) {
-    try {
-      this.logger.debug(`processing cache refresh for ${job.data}`);
-      let doc: DIDDocumentEntity;
-      if (this.configService.get<boolean>('DID_SYNC_MODE_FULL')) {
-        doc = await this.didService.addCachedDocument(job.data, true);
-      } else {
-        doc = await this.didService.incrementalRefreshCachedDocument(job.data);
-      }
+  @OnGlobalQueueFailed()
+  async onFailed(jobId: number) {
+    const job = await this.queue.getJob(jobId);
+    this.logger.debug(`Failed ${job.name} document ${job.data}`);
+  }
```

Comment:
@JGiter why use listeners for Global events, rather than local events?
I found this documentation discussing this but it's not clear to me why we would want Global listeners in our case https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#global-events
May Global is better than Local, it's just not clear to me why this would be.

Reply:
I registered listeners globaly becasue I am uncertaing about Redis configuration and afraid that events can be lost somehow. Perhaps it doesn't really make sence. I'll set listeners locally